packages/components-library/tests/unit/filters/ChangeFilters.spec.ts

```diff
@@ -0,0 +1,112 @@
+        collapsed: true,
+        bulkOperation: true,
+        options: [{ value: 'red', text: 'Red' }, { value: 'green', text: 'Green' }, { value: 'blue', text: 'Blue' }],
+        compound: 'compound-title',
+        type: 'checkbox-filter'
+      }],
+      value: ['search']
+    }
+  }
+
+  beforeEach(() => {
+    wrapper = mount(ChangeFilters, options)
+  })
+
+  it('can add a filter', (done) => {
+    wrapper.find('button.dropdown-toggle').trigger('click')
+    wrapper.vm.$nextTick(() => {
+      wrapper.findAll('.custom-control-input').at(1).trigger('click')
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper.emitted().input[0]).toEqual([['search', 'compound-title', 'string', 'checkbox']])
+        done()
+      })
+    })
+  })
+
+  it('can remove a filter', (done) => {
+    wrapper.find('button.dropdown-toggle').trigger('click')
+    wrapper.vm.$nextTick(() => {
+      wrapper.findAll('.custom-control-input').at(0).trigger('click')
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper.emitted().input[0]).toEqual([[]])
+        done()
+      })
+    })
+  })
+
+  it('can select all and deselect all with compounds', (done) => {
+    wrapper.find('button.dropdown-toggle').trigger('click')
+    wrapper.vm.$nextTick(() => {
+      // Select all
+      wrapper.findAll('.custom-control-input').at(1).trigger('click')
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper.emitted().input[0]).toEqual([['search', 'compound-title', 'string', 'checkbox']])
+        // Deselect all
+        wrapper.findAll('.custom-control-input').at(1).trigger('click')
+        wrapper.vm.$nextTick(() => {
+          expect(wrapper.emitted().input[1]).toEqual([['search']])
+          done()
+        })
+      })
+    })
+  })
+
+  it('can select and deselect sub-compounds', (done) => {
+    wrapper.find('button.dropdown-toggle').trigger('click')
+    wrapper.vm.$nextTick(() => {
+      // Select all sub-compounds (with all sub-compounds selected compound parent also gets selected)
+      wrapper.findAll('.custom-control-input').at(2).trigger('click')
+      wrapper.findAll('.custom-control-input').at(3).trigger('click')
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper.emitted().input[1]).toEqual([['compound-title', 'search', 'string', 'checkbox']])
+
+        // partial selection of sub-compounds
+        wrapper.findAll('.custom-control-input').at(2).trigger('click')
+        wrapper.vm.$nextTick(() => {
+          expect(wrapper.emitted().input[2]).toEqual([['search', 'checkbox']])
+
+          // deselect all sub-compounds
+          wrapper.findAll('.custom-control-input').at(3).trigger('click')
+          wrapper.vm.$nextTick(() => {
+            expect(wrapper.emitted().input[3]).toEqual([['search']])
+            done()
+          })
+        })
+      })
+    })
+  })
+
+  it('can make a array unique', () => {
+    // @ts-ignore
+    expect(['a', 'b', 'b', 'c', 'c', 'c'].filter(wrapper.vm.unique)).toEqual(['a', 'b', 'c'])
+  })
```

Comment:
Rather than testing internal component functions, wouldn't it be better to check the rendered output for double values where you wouldn't expect it? From vue-test-utils:
```
Instead, we recommend writing tests that assert your component's public interface, and treat its internals as a black box.
 ```

Reply:
yes, why is unique function part of this component ?